editoast/src/views/train_schedule/mod.rs

```diff
@@ -393,29 +403,88 @@ async fn create_backend_request_payload(
     })
 }
 
-fn process_simulation_response(
-    response_payload: SimulationResponse,
+pub fn process_simulation_response(
+    SimulationResponse {
+        base_simulations,
+        eco_simulations,
+        speed_limits,
+        electrification_conditions,
+        ..
+    }: SimulationResponse,
 ) -> Vec<SimulationOutputChangeset> {
     let mut simulation_outputs = Vec::new();
+    let electrification_conditions_iter = if electrification_conditions.is_empty() {
+        vec![None; base_simulations.len()]
+    } else {
+        electrification_conditions
+            .iter()
+            .map(|ec| Some(ec.clone()))
+            .collect()
+    };
     for (base_simulation, eco_simulation, speed_limits, electrification_conditions) in izip!(
-        response_payload.base_simulations,
-        response_payload.eco_simulations,
-        response_payload.speed_limits,
-        response_payload.electrification_conditions
+        base_simulations,
+        eco_simulations,
+        speed_limits,
+        electrification_conditions_iter
     ) {
+        let eco_simulation = match eco_simulation {
+            Some(eco) => Some(Some(DieselJson(eco))),
+            None => Some(None),
+        };
         let simulation_output = SimulationOutputChangeset {
             id: None,
             mrsp: Some(speed_limits),
             base_simulation: Some(DieselJson(base_simulation)),
-            eco_simulation: Some(Some(DieselJson(eco_simulation))),
-            electrification_conditions: Some(electrification_conditions),
-            train_schedule_id: None, // To be filled once the train schedule is inserted
+            eco_simulation,
+            electrification_conditions,
+            train_schedule_id: Some(None), // To be filled once the train schedule is inserted
         };
         simulation_outputs.push(simulation_output);
     }
     simulation_outputs
 }
 
+#[derive(Debug, QueryableByName)]
+pub struct OutputSimulationTrainSchedule {
+    #[diesel(sql_type = BigInt)]
+    pub id: i64,
+    #[diesel(sql_type = Jsonb)]
+    pub mrsp: JsonValue,
+    #[diesel(sql_type = Jsonb)]
+    pub base_simulation: DieselJson<ResultTrain>,
+    #[diesel(sql_type = Nullable<Jsonb>)]
+    pub eco_simulation: Option<DieselJson<ResultTrain>>,
+    #[diesel(sql_type = Jsonb)]
+    pub electrification_conditions: JsonValue,
+    #[diesel(sql_type = Nullable<BigInt>)]
+    pub train_schedule_id: Option<i64>,
+    #[diesel(sql_type = Nullable<Double>)]
+    pub schedule_departure_time: Option<f64>,
+}
+
+pub fn simulation_outputs_from_timetable_id(
```

Comment:
Should be a static function of `OutputSimulationTrainSchedule`

Reply:
done